### ISSUE TITLE
Conda environment folders are not recognized as virtual environments

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pylance-release#5191]
### Discussed in https://github.com/microsoft/pylance-release/discussions/5143

<div type='discussions-op-text'>

<sup>Originally posted by **Ritesh313** November 24, 2023</sup>
I'm getting `"<module_name>"is not accessed` warnings from Pylance in my python file while importing all the stdlib modules.
The warning description just says: 
`conda_env_location/lib/python3.9/<module_name>.py" is overriding the stdlib module "<module_name>"`

I have my anaconda environment selected as my interpreter in VS code. I know it's just a warning and not interrupting my code, but I would just like to know if I have something setup wrong. Please let me know if I should provide some other information for this.

Modules which have this warning: datetime, random, typing, argparse
Platform: VS code
Environment: Conda
Python version: 3.9.7

Here's a screenshot of the warning:
<img width="1475" alt="image" src="https://github.com/microsoft/pylance-release/assets/36135489/f59ab3c3-fe40-46ae-96ae-d7da0fcb6905">

</div>

**Comment by rchiodo** [comment]
As a workaround you can disable the warning for now like so:

```jsonc 
    "python.analysis.diagnosticSeverityOverrides": {
        "reportShadowedImports": "none"
    }
```

**Comment by rchiodo** [comment]
I can reproduce this in a conda enviroment if I open the 'envs' folder:

![image](https://github.com/microsoft/pylance-release/assets/19672699/91b7262f-4bf6-4fdc-9127-28cb73c11074)

We don't detect conda environments as virtual environments because they don't have a `bin\activate` or `Scripts\activate` file. 

We could probably detect this situation, but I don't think this is very common. Most of the time the env folder is tucked away under the home directory.

**Comment by msegado** [comment]
> We could probably detect this situation, but I don't think this is very common. Most of the time the env folder is tucked away under the home directory.

I do know there are also up-and-coming tools like [Pixi](https://pixi.sh) that prefer to put Conda envs in a project subfolder, so it may become more common in the future. Pixi stores things in `{project_root}/.pixi/` though, so it's ignored by Pylance (which excludes `**/.*` paths by default) and doesn't cause issues.

For anyone else who likes to put Conda envs in a .gitignored subfolder of their projects (like me!), Pylance will happily ignore it as long as you name your subfolder something like `.env/` =)

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2024.4.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20244100-4-April-2024-PreRelease)
